### PR TITLE
Support update of Device Service

### PR DESCRIPTION
**Pull request by difince**
New command added ./edgex-cli deviceservice update
Device service could be updated in two ways:
- by providing a file that contains json representation of a single/list of Device Service(s)
- by using interactive mode. Device service with given name (--name) is loaded into
default editor, ready to be customized by the user.

Fix: https://github.com/edgexfoundry/edgex-cli/issues/249

Signed-off-by: Diana Atanasova <dianaa@vmware.com>

### CODE REVIEW

**Comment by difince** [comment]
How do you think the user experience will be using the interactive mode ?! 
IMO interactive mode improves the user experience and will be good to have it in create/update commands 
What are your opinions?

**Review by jpwhitemn — DISMISSED** [review]
@difince - LGTM.  Only issue is a minor one.  I would suggest providing something in deviceservice help that indicates use of --name and --file options.  I had to look at the PR and the code to figure out these flags and how to specify the JSON or how to trigger the editor to come up.

File: cmd/deviceservice/add/add.go [hunk 1]
```diff
@@ -1,35 +1,24 @@
...
 
 import (
-	"bytes"
 	"context"
 	"encoding/json"
 	"errors"
 	"fmt"
+	"html/template"
+
+	"github.com/edgexfoundry-holding/edgex-cli/cmd/deviceservice/update"
```
Comment (mkbhanda) [inline cmd/deviceservice/add/add.go:10]: holding drop ... ?

File: cmd/deviceservice/add/add.go [hunk 2]
```diff
@@ -38,19 +27,22 @@ var operState string
 var addrName string
 var file string
 
+const DeviceServicesTempl = `[{{range $ds := .}}` + update.DeviceSrviceTempl +
```
Comment (mkbhanda) [inline cmd/deviceservice/add/add.go:30]: update.DeviceSrviceTempl

File: cmd/deviceservice/add/add.go [hunk 3]
```diff
@@ -38,19 +27,22 @@ var operState string
...
+	`{{end}}]`
+
 func NewCommand() *cobra.Command {
 	cmd := &cobra.Command{
 		Use:   "add",
 		Short: "Add device service",
-		Long:  `Create device service described in the given JSON file or use the interactive mode with additional flags.`,
-		RunE: newDeviceServiceHandler,
+		Long:  `Create device service described in a given JSON file or use the interactive mode with additional flags.`,
```
Comment (mkbhanda) [inline cmd/deviceservice/add/add.go:37]: s/service/service(s)

File: pkg/editor/interactive.go [hunk 4]
```diff
@@ -86,3 +88,28 @@ func CaptureInputFromEditor(template []byte) ([]byte, error) {
...
+		return nil, err
+	}
+
+	return CaptureInputFromEditor(buff.Bytes())
+}
+
+// isLastElementOfSlice is a function which is used in HTML templates to determine the last element in a slice
+func IsLastElementOfSlice(index int, lenght int) bool {
+	return index == lenght-1
+}
```
Comment (mkbhanda) [inline pkg/editor/interactive.go:115]: newline

**Review by mkbhanda — COMMENTED** [review]
minor comments

File: cmd/deviceservice/add/add.go (hunk 1 above)
Comment (difince) [reply cmd/deviceservice/add/add.go:10]: good catch up! I haven't noticed it at all. 
It looks to me that moving out of holding haven't happened entirely correctly. 
Will open new issue about this.

**Comment by difince** [comment]
@mkbhanda, @jpwhitemn Thank you for your comments 
I tried to create explanatory helper messages ... but I am open for better ideas if any

File: cmd/deviceservice/add/add.go (hunk 1 above)
Comment (difince) [reply cmd/deviceservice/add/add.go:10]: @jpwhitemn any ideas?

**Review by mkbhanda — APPROVED** [review]
Spoke to Diana, we merge this PR and the other one that this depends on will address a few typos/nits I caught.

**Pull request merged**